thumbor_aws/s3_client.py

```diff
@@ -83,20 +83,21 @@ def file_acl(self) -> str:
 
     @property
     def session(self) -> AioSession:
-        """Singleton Session used for connecting with AWS"""
-        if self.__session is None:
-            self.__session = get_session()
-        return self.__session
-
-    def get_client(self) -> AioBaseClient:
-        """Gets a connected client to use for S3"""
-        return self.session.create_client(
-            "s3",
-            region_name=self.region_name,
-            aws_secret_access_key=self.secret_access_key,
-            aws_access_key_id=self.access_key_id,
-            endpoint_url=self.endpoint_url,
-        )
+        """Session used for connecting with AWS"""
+        return get_session()
+
+    async def get_client(self) -> AioBaseClient:
+        """Singleton client for S3"""
+        if self.s3_client is None:
+            client = self.session.create_client(
+                "s3",
+                region_name=self.region_name,
+                aws_secret_access_key=self.secret_access_key,
+                aws_access_key_id=self.access_key_id,
+                endpoint_url=self.endpoint_url,
+            )
+            self.s3_client = await client.__aenter__()
```

Comment:
We should be calling __ aexit__ shouldn't we? I think the correct way here is to assing self.s3_client to client and have the rest of the codebase do:

```python
with s3_client as client:
    # Whatever you need to do!
```

This way we ensure proper disposition of resources, right?

Reply:
We might not need all the changes I've made; the session object should be enough to reuse the same HTTP connections.

I would need @oliverschewe help to validate this.

Reply:
@heynemann I've updated the PR, to only reuse the S3Client object, but to still keep the `with s3_client as client` usage.